005a-SIP2_checkout_branch

This patch affects SIP2 checkout to override branch with the AO (institution id) field.
Since SIP2 also uses C4::Context this patch has to modify C4::Context->userenv->{'branch'}

diff --git a/C4/SIP/Sip/C4/SIP/Sip/MsgType.pm b/C4/SIP/Sip/MsgType.pm
index 787449b..f21244a 100644
--- a/C4/SIP/Sip/MsgType.pm
+++ b/C4/SIP/Sip/MsgType.pm
@@ -14,6 +14,7 @@ use Sys::Syslog qw(syslog);
 use C4::SIP::Sip qw(:all);
 use C4::SIP::Sip::Constants qw(:all);
 use C4::SIP::Sip::Checksum qw(verify_cksum);
+use C4::Context;
 
 use Data::Dumper;
 use CGI qw ( -utf8 );
@@ -499,6 +500,11 @@ sub handle_checkout {
     $item_id   = $fields->{ (FID_ITEM_ID) };
     my $fee_ack = $fields->{ (FID_FEE_ACK) };
 
+    # Override Context branch with AO (Field institution ID) from SIP
+    # this allows proper handling of branch checkout
+    $inst = $fields->{(FID_INST_ID)};
+    C4::Context->userenv->{'branch'} = $inst;
+
     if ( $no_block eq 'Y' ) {
 
         # Off-line transactions need to be recorded, but there's
